### PR TITLE
Updated python scripts in AutomatedTesting to split PhysX Collider into Primitive Collider and Mesh Collider

### PR DESCRIPTION
**Pull request by moraaar**
**Note**
This is part of the work to separate PhysX Collider into 2 components: https://github.com/o3de/o3de/pull/14725. Merging into the staging branch `MeshColliderComponent_Staging`.

Signed-off-by: moraaar <moraaar@amazon.com>

## What does this PR do?

This are the test areas where colliders were found and updated: **Physics, Prefabs, Terrain, largeworlds dyn_veg, EditorPythonTestTools and EditorPythonBindings.**

- Updated `"PhysX Collider"` with `"PhysX Primitive Collider"` if they were using primitive or with `"PhysX Mesh Collider"` if they were using PhysX Asset.
- When using `PhysX Mesh Collider` the calls to `SetShapeType` were removed since it's not applicable anymore.
- When using `PhysX Mesh Collider` updated the path to set the physx asset from `"Shape Configuration|Asset|PhysX Mesh"` to `"Physics Asset|PhysX Mesh"`
- Split `editor_physx_collider.py` into `editor_physx_primitive_collider.py` and `editor_physx_mesh_collider.py`.
- Split `PhysX_Collider_Component_CRUD` into `PhysX_Primitive_Collider_Component_CRUD` and `PhysX_Mesh_Collider_Component_CRUD`.

## How was this PR tested?

AR run pass, which runs all the smoke and main python tests.
Tests modified that are part of periodic and sandbox have been run locally and passed, exception being Physics ones because they were failing before this change.

### CODE REVIEW

**Review by cgalvan — APPROVED** [review]

**Review by jckand-amzn — APPROVED** [review]

**Comment by jckand-amzn** [comment]
Thanks for checking the Periodic tests as well!

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderAnotherPrefab.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def CreatePrefab_UnderAnotherPrefab():
     entity = inner_instance.get_direct_child_entities()[0]
     # We track if that is the same entity by checking the name and if it still contains the component that we created before
     assert entity.get_name() == "TestEntity", f"Entity name inside inner_prefab doesn't match the original name, original:'TestEntity' current:'{entity.get_name()}'"
-    assert entity.has_component("PhysX Collider"), "Entity name inside inner_prefab doesn't have the collider component it should"
+    assert entity.has_component("PhysX Primitive Collider"), "Entity name inside inner_prefab doesn't have the collider component it should"
```
Comment (FuzzyCarterAWS) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderAnotherPrefab.py:54]: NIT: Why not import the PHYSX_PRIMITIVE_COLLIDER from Editor Python Test Tools so we can eliminate passing hand typed/pasted strings around while getting the benefit of autocomplete?

`from editor_python_test_tools.consts.physics import PHYSX_PRIMITIVE_COLLIDER`
`assert entity.has_component(PHYSX_PRIMITIVE_COLLIDER), "Entity name inside inner_prefab doesn't have the collider component it should"`

**Review by FuzzyCarterAWS — APPROVED** [review]
This looks fantastic.

Biggest piece of feedback is to use the consts we've built in our libraries instead of continuing to copy/paste/type the same strings over and over in our tests.

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_physx_mesh_collider.py [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+        """
+        _validate_xyz_is_float(x, y, z, f"One of the rotation inputs are not a float - X: {x}, Y: {y}, Z {z}")
+
+        rotation = math.Quaternion()
+        rotation.SetFromEulerDegrees(math.Vector3(x, y, z))
+        self.component.set_component_property_value(self.Path.ROTATION, rotation)
+
+    def get_rotation(self) -> math.Quaternion:
+        """
+        Property Type, Default Visibility - ('Vector3', 'Visible')
```
[the remark this answers is not in the document]
Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_physx_mesh_collider.py:139]: Thanks, I'll change it.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py [hunk 3]
```diff
@@ -27,20 +27,20 @@ def CreatePrefab_UnderChildEntityOfAnotherPrefab():
     INNER_PREFAB_FILE_NAME = Path(__file__).stem + INNER_PREFAB_NAME
     PARENT_ENTITY_NAME = 'ParentEntity'
     CHILD_ENTITY_NAME = 'ChildEntity'
-    PHYSX_COLLIDER_NAME = 'PhysX Collider'
+    PHYSX_PRIMITIVE_COLLIDER_NAME = 'PhysX Primitive Collider'
```
[the remark this answers is not in the document]
Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderChildEntityOfAnotherPrefab.py:30]: Good call! I have a few follow-up tasks to do, I'll make a note of it and do this after the work is merged to development.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderAnotherPrefab.py (hunk 1 above)
Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderAnotherPrefab.py:54]: Thanks! I'll clean this up in a follow-up PR.

File: AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_physx_mesh_collider.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        """
+        Property Type, Default Visibility - ('Quaternion', 'Visible')
+
+        Used to set the PhysX Mesh Collider's Rotation Property.
+        """
+        _validate_xyz_is_float(x, y, z, f"One of the rotation inputs are not a float - X: {x}, Y: {y}, Z {z}")
+
+        rotation = math.Quaternion()
+        rotation.SetFromEulerDegrees(math.Vector3(x, y, z))
+        self.component.set_component_property_value(self.Path.ROTATION, rotation)
```
[the remark this answers is not in the document]
Comment (FuzzyCarterAWS) [reply AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_physx_mesh_collider.py:135]: Ahh, that code wasn't working for me then and I left it as a starting point.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderAnotherPrefab.py (hunk 1 above)
Comment (moraaar) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/create_prefab/CreatePrefab_UnderAnotherPrefab.py:54]: Follow up PR doing this clean up: https://github.com/o3de/o3de/pull/14955

**Pull request merged**